compass/sign/RemoteSignatureSource.java

```diff
@@ -40,15 +48,66 @@ public RemoteSignatureSource(String uri,
                                String trustCertCollectionFilePath,
                                String clientCertChainFilePath,
                                String clientPrivateKeyFilePath) throws SSLException {
-    this(NettyChannelBuilder
+    this(OkHttpChannelBuilder
         .forTarget(uri)
         .useTransportSecurity()
         .idleTimeout(5, TimeUnit.SECONDS)
-        .sslContext(
-            buildSslContext(trustCertCollectionFilePath, clientCertChainFilePath, clientPrivateKeyFilePath))
+        .sslSocketFactory(ClientCertSslSocketFactory(trustCertCollectionFilePath, clientCertChainFilePath, clientPrivateKeyFilePath))
         .build());
   }
 
+  private static PrivateKey createPrivateKeyFromPemFile(final String keyFileName) throws IOException, InvalidKeySpecException, NoSuchAlgorithmException {
+      PemReader pemReader = new PemReader(new FileReader(keyFileName));
+      PemObject pemObject = pemReader.readPemObject();
+      byte[] pemContent = pemObject.getContent();
+      pemReader.close();
+      PKCS8EncodedKeySpec encodedKeySpec = new PKCS8EncodedKeySpec(pemContent);
+      KeyFactory keyFactory = KeyFactory.getInstance("RSA");
+      PrivateKey privateKey = keyFactory.generatePrivate(encodedKeySpec);
+      return privateKey;
+  }
+
+  public static SSLSocketFactory ClientCertSslSocketFactory(String trustCertCollectionFilePath,
+                                                            String clientCertChainFilePath,
+                                                            String clientPrivateKeyFilePath ) throws RuntimeException {
```

Comment:
I thought the `public` was necessary since the method is actually invoked from the ChannelBuilder each time a new connection attempt needs to be made.